app/lib/actions/katello/event_queue/poller_thread.rb

```diff
@@ -26,12 +26,15 @@ def close
         def run_event(event)
           @logger.debug("event_queue_event: #{event.event_type}, #{event.object_id}")
           ::User.as_anonymous_admin do
-            ::Katello::EventQueue.event_class(event.event_type).new(event.object_id).run
+            ::Katello::EventQueue.create_instance(event).run
           end
         rescue => e
           @logger.error("event_queue_error: #{event.event_type}, #{event.object_id}")
           @logger.error(e.message)
           @logger.error(e.backtrace.join("\n"))
+
+          rescheduled = ::Katello::EventQueue.reschedule_event(event)
+          @logger.warn("event_queue_rescheduled: type=#{event.event_type} object_id=#{event.object_id}") if rescheduled
```

Comment:
I think we should likely catch a specific error such that:

* we don't reschedule an event for some unexpected error that is likely to not be recoverable (forever and ever)
* we dont' reschedule an event that isn't meant to be rescheduled

My thinking is that below, we explicitly check for a 'lock' exception, and then throw a specific 'reschedule' exception that is caught here. 

Thoughts?

Reply:
The way the code is written right now won't reschedule every failed event run - only those where the concrete class implements the `retry_seconds` seconds method.

Are you saying that we should tightly control the retry behavior for particular modes of failure that we determine rather than just retrying if `retry_seconds` is implemented (which would retry on any failure)?

Reply:
yeah, i think we should only retry on a specific failure (a lock exception) if possible